cpp/ql/lib/experimental/semmle/code/cpp/semantic/SemanticExprSpecific.qll

```diff
@@ -0,0 +1,301 @@
+/**
+ * C++-specific implementation of the semantic interface.
+ */
+
+private import cpp as Cpp
+private import semmle.code.cpp.ir.IR as IR
+private import Semantic
+private import experimental.semmle.code.cpp.rangeanalysis.Bound as IRBound
+private import semmle.code.cpp.controlflow.IRGuards as IRGuards
+
+module SemanticExprConfig {
+  class Location = Cpp::Location;
+
+  class Expr = IR::Instruction;
+
+  string exprToString(Expr e) { result = e.toString() }
```

Comment:
Good question. I think that's what I wound up doing for all the other interfaces. Most of the other predicates in the interface between the semantic layer and the language are non-member predicates, so that they aren't visible on the semantic objects that get exposed to the analysis. For `toString()` and `getLocation()`, though, we wind up exposing them anyway, so I don't think there's a need to hide them. Fixed.